packages/flutter/lib/src/painting/placeholder_span.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui show ParagraphBuilder, PlaceholderAlignment;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/services.dart';
+
+import 'basic_types.dart';
+import 'inline_span.dart';
+import 'text_style.dart';
+import 'text_painter.dart';
+import 'text_span.dart';
+
+/// An immutable placeholder that is embedded inline within text.
+///
+/// [PlaceholderSpan] represents an abstract generic placeholder. [WidgetSpan] should be
+/// used instead to specify the contents of the placeholder. A [PlaceholderSpan]
```

Comment:
We can't tell people to use WidgetSpan in general, they might not be using Widgets. For example, consider if you're using the TextPainter API directly in a CustomPaint.